### PR TITLE
Generate Region Plots Directly

### PR DESCRIPTION
**Pull request by pshriwise**
# Description

A continuation of #2648, these changes allow regions to be plotted directly without need for a cell definition. A simple region can now be plotted with the following:

```python 
import openmc
from matplotlib import pyplot as plt

sphere = openmc.Sphere()
x_plant = openmc.XPlane()
hemisphere = -sphere & +x_plane
hemisphere.plot()
plt.show()
```

hemisphere

There was a little more nuance here as the `color_by`, `color`, `legend`, and `legend_kwargs` arguments no longer have meaning at this level of geometry definition, so I've addressed those as best I could for now.

# Other 

Also cleaned up the `openmc.Cell.plot` docstring a bit.

# Checklist

- [x] I have performed a self-review of my own code
~- [ ] I have run [clang-format](https://docs.openmc.org/en/latest/devguide/styleguide.html#automatic-formatting) (version 15) on any C++ source files (if applicable)~
- [x] I have followed the [style guidelines](https://docs.openmc.org/en/latest/devguide/styleguide.html#python) for Python source files (if applicable)
- [x] I have made corresponding changes to the documentation (if applicable)
- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)

### CODE REVIEW

**Comment by shimwell** [comment]
Nice PR. We might be able to plot everything thing soon :rocket: 
Region - this PR
Cell - done
Geometry - done
Universe - done
Model - PR #2863

**Review by paulromano — APPROVED** [review]
Thanks!

**Comment by shimwell** [comment]
Looks like that extra assert in the testing picked up an error in the latest CI run.

File: tests/unit_tests/test_region.py [hunk 1]
```diff
@@ -241,3 +241,13 @@ def test_invalid_operands():
...
+
+def test_plot():
+    # Create region and plot
+    region = -openmc.Sphere() & +openmc.XPlane()
+    c_before = openmc.Cell()
+    region.plot()
+
+    # Ensure that calling plot doesn't affect cell ID space
+    c_after = openmc.Cell()
+    assert c_after.id + 1 == c_before.id
```
Comment (shimwell) [inline tests/unit_tests/test_region.py:253]:
```suggestion
    assert c_after.id - 1 == c_before.id
```
should this be a - instead of a +

Comment (paulromano) [reply tests/unit_tests/test_region.py:253]: Oops that's my fault! Thanks for the fix

**Pull request merged**